doc/rule-dsl.md

```diff
@@ -0,0 +1,510 @@
+
+#### Property: destination <a id="destination"></a>
+
+Every rule corresponds to some destination microservice identified by a `destination` field in the
+rule. For example, all rules that apply to calls to the "reviews" microservice will include the following field.
+
+```yaml
+destination: reviews.default.svc.cluster.local
+```
+
+#### Property: precedence <a id="precedence"></a>
+
+The order of evaluation of rules corresponding to a given destination, when there is more than one, can be specified 
+by setting the `precedence` field of the rule. 
+
+```yaml
+destination: reviews.default.svc.cluster.local
+precedence: 1
+```
+
+The `precedence` field is an optional integer value, 0 by default.
+Rules with higher precedence values are evaluated first.
+If there is more than one rule with the same precedence value, the order of evaluation is undefined.
+
+For further details, see [Route Rule Evaluation](#route-rule-evaluation).
+
+#### Property: match <a id="match"></a>
+
+Rules can optionally be qualified to only apply to requests that match some specific criteria such
+as a specific request source and/or headers. An optional `match` field is used for this purpose.
+
+The `match` field is an object with the following nested fields:
+
+* `source`
+* `source_tags`
+* `http`
+* `tcp`
+* `udp`
+
+#### Property: match.source <a id="match-source"></a>
+
+The `source` field is used to qualify a rule to only apply to requests from a specific caller.
+For example,
+the following `match` clause qualifies the rule to only apply to calls from the "reviews" microservice.
+
+```yaml
+destination: ratings.default.svc.cluster.local
+match:
+  source: reviews.default.svc.cluster.local
+```
+
+#### Property: match.source_tags <a id="match-source_tags"></a>
+
+The `source_tags` field can be used to further qualify a rule to only apply to specific instances of a
+calling service.
+For example,
+the following `match` clause refines the previous example to qualify the rule to only 
+apply to calls from version "v2" of the "reviews" microservice.
+
+```yaml
+destination: ratings.default.svc.cluster.local
+match:
+  source: reviews.default.svc.cluster.local
+  source_tags:
+    version: v2
+```
+
+#### Property: match.http <a id="match-http"></a>
+
+The `http` field is a set of one or more property-value pairs where each property is an HTTP header name
+and the corresponding value is one of the following:
+
+1. `exact: value`, where the header value must match `value` exactly
+2. `prefix: value`, where `value` must be a prefix of the header value
+3. `regex: value`, where `value` is a regular expression that matches the header value
+
+For example, the following rule will only apply
+to an incoming request if it includes a "Cookie" header that contains the substring "user=jason".
+
+```yaml
+destination: reviews.default.svc.cluster.local
+match:
+  http:
+    Cookie:
+      regex: "^(.*?;)?(user=jason)(;.*)?$"
+```
+
```

Comment:
You might also want to include the Uri field as an example (the most common one), with a prefix match.. like

```yaml
match:
   http:
     Uri:
       prefix: /svc/reviews/
```